core/src/main/java/org/apache/carbondata/core/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,171 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.carbondata.common.exceptions.sql.MalformedCarbonCommandException;
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.util.CustomIndex;
+
+/**
+ * GeoHash default implementation
+ */
+public class GeoHashImpl implements CustomIndex<Long, String, List<Long[]>> {
+  /**
+   * Initialize the geohash index handler instance.
+   * @param handlerName
+   * @param properties
+   * @throws Exception
+   */
+  @Override
+  public void init(String handlerName, Map<String, String> properties) throws Exception {
+    String options = properties.get(CarbonCommonConstants.INDEX_HANDLER);
+    if (options == null || options.isEmpty()) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid.", CarbonCommonConstants.INDEX_HANDLER));
+    }
+    options = options.toLowerCase();
+    if (!options.contains(handlerName.toLowerCase())) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s is not present.",
+                      CarbonCommonConstants.INDEX_HANDLER, handlerName));
+    }
+
+    String commonKey = CarbonCommonConstants.INDEX_HANDLER + "." + handlerName + ".";
+    String TYPE = commonKey + "type";
+    String SOURCE_COLUMNS = commonKey + "sourcecolumns";
+    String SOURCE_COLUMN_TYPES = commonKey + "sourcecolumntypes";
+    String TARGET_DATA_TYPE = commonKey + "datatype";
+    String ORIGIN_LATITUDE = commonKey + "originlatitude";
+    String MIN_LONGITUDE = commonKey + "minlongitude";
+    String MAX_LONGITUDE = commonKey + "maxlongitude";
+    String MIN_LATITUDE = commonKey + "minlatitude";
+    String MAX_LATITUDE = commonKey + "maxlatitude";
+    String GRID_SIZE = commonKey + "gridsize";
+    String CONVERSION_RATIO = commonKey + "conversionratio";
+
+
+    String sourceColumnsOption = properties.get(SOURCE_COLUMNS);
+    if (sourceColumnsOption == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property is not specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+    }
+
+    if (sourceColumnsOption.split(",").length != 2) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must have 2 columns.",
+                      CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+    }
+
+    String type = properties.get(TYPE);
+    if (type != null && !CarbonCommonConstants.GEOHASH.equalsIgnoreCase(type)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be %s for this class.",
+                      CarbonCommonConstants.INDEX_HANDLER, TYPE, CarbonCommonConstants.GEOHASH));
+    }
+
+    properties.put(TYPE, CarbonCommonConstants.GEOHASH);
```

Comment:
TYPE is already there in properties, no need to put again

Reply:
Agreed. Modified.